prow/plugins/plugins.go

```diff
@@ -138,9 +135,41 @@ type PluginClient struct {
 	Logger *logrus.Entry
 }
 
-type PluginAgent struct {
-	PluginClient
+func NewAgent(configAgent *config.Agent, pluginConfigAgent *ConfigAgent, clientAgent *ClientAgent, logger *logrus.Entry) Agent {
+	return Agent{
+		GitHubClient: clientAgent.GitHubClient,
+		KubeClient:   clientAgent.KubeClient,
+		GitClient:    clientAgent.GitClient,
+		SlackClient:  clientAgent.SlackClient,
+		Config:       configAgent.Config(),
+		PluginConfig: pluginConfigAgent.Config(),
+		Logger:       logger,
+	}
+}
+
+func (a *Agent) OwnersClient() *repoowners.Client {
```

Comment:
Similarly to `commentpruner`, the `repoowners` client caches to reduce API token usage. 
There should only be one `repoowners` client shared by all handlers for all events.